src/core/sio1-server.cc

```diff
@@ -57,5 +65,60 @@ void PCSX::SIO1Server::startServer(uv_loop_t* loop, int port) {
 
 void PCSX::SIO1Server::stopServer() {
     m_serverStatus = SIO1ServerStatus::SERVER_STOPPING;
+    g_emulator->m_counters->m_pollSIO1 = false;
     m_fifoListener.stop();
 }
+
+PCSX::SIO1Client::SIO1Client() : m_listener(g_system->m_eventBus) {
+    m_listener.listen<Events::SettingsLoaded>([this](const auto& event) {
+        if (g_emulator->settings.get<Emulator::SettingDebugSettings>().get<Emulator::DebugSettings::SIO1Client>() &&
+            (m_clientStatus != SIO1ClientStatus::CLIENT_STARTED)) {
+                startClient(std::string_view(g_emulator->settings.get<Emulator::SettingDebugSettings>().get<Emulator::DebugSettings::SIO1ClientHost>().value),
+                        g_emulator->settings.get<Emulator::SettingDebugSettings>().get<Emulator::DebugSettings::SIO1ClientPort>());
+        }
+    });
+    m_listener.listen<Events::Quitting>([this](const auto& event) {
+        if (m_clientStatus == SIO1ClientStatus::CLIENT_STARTED) stopClient();
+    });
+}
+
+void PCSX::SIO1Client::startClient(std::string_view address, unsigned port) {
+    if (m_clientStatus == SIO1ClientStatus::CLIENT_STARTED) {
+        throw std::runtime_error("Client already started");
+    }
+    m_address = address;
+    m_port = port;
+    auto& emuSettings = PCSX::g_emulator->settings;
+    auto& debugSettings = emuSettings.get<Emulator::SettingDebugSettings>();
+    auto SIO1ModeSettings = debugSettings.get<Emulator::DebugSettings::SIO1ModeSetting>().value;
+    if (SIO1ModeSettings == Emulator::DebugSettings::SIO1Mode::Raw) {
+        g_emulator->m_sio1->m_sio1Mode = SIO1::SIO1Mode::Raw;
+    } else {
+        g_emulator->m_sio1->m_sio1Mode = SIO1::SIO1Mode::Protobuf;
+        g_emulator->m_counters->m_pollSIO1 = true;
```

Comment:
No, we don't need the state machine polling in Raw mode because we are writing the data directly into the sio1fifo now and when branchtest checks for an interrupt, in raw mode, if there is data, it will schedule one. So no need for polling state machine in Raw mode